tests/virtualization/xen/virtmanager.pm

```diff
@@ -29,28 +29,29 @@ sub run {
 
     x11_start_program 'virt-manager', target_match => 'virt-manager';
 
-    assert_screen "virt-manager_add-connection";
-    send_key 'spc';
-    send_key 'down';
-    send_key 'down';
-    send_key 'spc';
-    save_screenshot;    # XEN selected
-    send_key 'tab';
-    send_key 'spc';
-    save_screenshot;    # Connect to remote host ticked
-    send_key 'tab';
-    send_key 'tab';
-    type_string 'root';
-    save_screenshot;    # root written
-    send_key 'tab';
-    type_string "$hypervisor";
-    save_screenshot;    # $hypervisor written
-    send_key 'tab';
-    send_key 'spc';
-    save_screenshot;    # autoconnect ticked
-    send_key 'ret';
-    assert_screen "virt-manager_connected";
+    if (check_screen('virt-manager_add-connection', 15)) {
```

Comment:
Please use a multi-tag `assert_screen` with `match_has_tag` instead of `check_screen` with non-zero timeout to prevent introducing any timing dependant behaviour, to save test execution time as well as state more explicitly from the testers point of view what are the expected alternatives. For example:

```perl
assert_screen([qw(yast2_console-finished yast2_missing_package)]);
if (match_has_tag('yast2_missing_package')) {
    send_key 'alt-o';  # confirm package installation
    assert_screen 'yast2_console-finished';
}
```